code-challenges.test.js

```diff
@@ -15,35 +15,149 @@
 
 // a) Create a test with an expect statement using the variable provided. HINT: Check out this resource: https://jestjs.io/docs/expect#expectarraycontainingarray
 
-const colors1 = ["purple", "blue", "green", "yellow", "pink"]
-// Expected output example (can be a different order): ["yellow", "blue", "pink", "green"]
-const colors2 = ["chartreuse", "indigo", "periwinkle", "ochre", "aquamarine", "saffron"]
-// Expected output example (can be a different order): ["saffron", "aquamarine", "periwinkle", "indigo", "ochre"]
-
+// Create a test with describe, it and two expects with arrayContainings
+// make it fail
+
+describe("delFirstShuffle", () => {
+    it("takes in an array, removes the first item from the array and shuffles the remaining content.", () => {
+        const colors1 = ["purple", "blue", "green", "yellow", "pink"]
+        // Expected output example (can be a different order): ["yellow", "blue", "pink", "green"]
+        const colors2 = ["chartreuse", "indigo", "periwinkle", "ochre", "aquamarine", "saffron"]
+        // Expected output example (can be a different order): ["saffron", "aquamarine", "periwinkle", "indigo", "ochre"]
+        expect(delFirstShuffle(colors1)).toEqual(expect.arrayContaining(["yellow", "blue", "pink", "green"]))
+        expect(delFirstShuffle(colors2)).toEqual(expect.arrayContaining(["saffron", "aquamarine", "periwinkle", "indigo", "ochre"]))
+    })
+})
+
+// RED 
+// FAIL  ./code-challenges.test.js
+// delFirstShuffle
+//   ✕ takes in an array, removes the first item from the array and shuffles the remaining content. (1 ms)
+// ● delFirstShuffle › takes in an array, removes the first item from the array and shuffles the remaining content.
+//   ReferenceError: delFirstShuffle is not defined
 
 // b) Create the function that makes the test pass.
 
+// I found a common and accurate way to shuffle arrays through an algorithm called the Fisher-Yates Shuffle
+// Create a function named delFirstShuffle that takes in array as its param
+// Shift the first element off the array
+// Create a for loop to iterate through our array
+    // create a local variable named rIndex that contains math random times i rounded down 
+    // push array at rIndex into our original array
+    // splice our array starting at rIndex and deleting one item
+// return the array
+// pass test
+
+const delFirstShuffle = (array) => {
+    array.shift()
+    for(let i = 0; i < array.length; i++) {
+        rIndex = Math.floor(Math.random() * i)
+        array.push(array[rIndex])
+        array.splice(rIndex, 1)
```

Comment:
Very cool, removing items out of the old array as you move them into the new array guarantees no duplicates. Awesome.